src/server-config.ts

```diff
@@ -75,12 +76,27 @@ const checkPassword = (password: string): boolean => {
   return containsLetter.test(password) && containsDigit.test(password);
 };
 
-export const startServer = async (): Promise<void> => {
+export const startServer = async (): Promise<ApolloServer> => {
+  const path: string = process.env.TEST === 'OK' ? './test.env' : './.env';
+  dotenv.config({ path });
+
   const server = new ApolloServer({ typeDefs, resolvers });
 
-  await createConnection();
+  await createConnection({
+    type: 'postgres',
+    host: process.env.PS_HOST,
+    port: +(process.env.PS_PORT ?? 5432),
+    username: process.env.PS_USER,
+    password: process.env.PS_PASSWORD,
+    database: process.env.PS_DATABASE,
+    synchronize: true,
+    logging: false,
+    entities: [User]
+  });
 
   server.listen(4000).then((response) => {
     console.log(`Server ready at ${response.url}`);
   });
+
+  return server;
```

Comment:
```suggestion
  const response = await server.listen(4000)
  console.log(`Server ready at ${response.url}`);

  return server;
```

Essa sugestão é pra que ele retorne o `server` apenas depois que o `listen` deu certo. O `listen` tbm retorna Promise, e isso quer dizer que ele pode falhar (um exemplo de falha: se a porta 4000 já estiver sendo usad para outra coisa). Da forma anterior, pode ser que o `listen` falhasse mas que essa função `startServer` retornasse como se ele não tivesse falhado. Ficou claro?

Reply:
Entendi, obrigada pela sugestão